components/widgets/menu/menu_items/menu_cloud_trial.tsx

```diff
@@ -2,53 +2,127 @@
 
 import React from 'react';
-import {FormattedMessage} from 'react-intl';
-import {useSelector} from 'react-redux';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
 
 import UpgradeLink from 'components/widgets/links/upgrade_link';
+import TrialBenefitsModal from 'components/trial_benefits_modal/trial_benefits_modal';
 
+import {DispatchFunc} from 'mattermost-redux/types/actions';
 import {GlobalState} from 'mattermost-redux/types/store';
 import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {cloudFreeEnabled} from 'mattermost-redux/selectors/entities/preferences';
+
+import {openModal} from 'actions/views/modals';
 
 import {getRemainingDaysFromFutureTimestamp} from 'utils/utils';
-import {TrialPeriodDays} from 'utils/constants';
+import {TrialPeriodDays, ModalIdentifiers} from 'utils/constants';
 
 import './menu_item.scss';
+import CloudStartTrialBtn from 'components/cloud_start_trial/cloud_start_trial_btn';
 
 type Props = {
     id: string;
 }
-const MenuCloudTrial: React.FC<Props> = ({id}: Props) => {
+const MenuCloudTrial = ({id}: Props) => {
     const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
     const license = useSelector(getLicense);
+    const dispatch = useDispatch<DispatchFunc>();
+    const {formatMessage} = useIntl();
 
     const isCloud = license?.Cloud === 'true';
     const isFreeTrial = subscription?.is_free_trial === 'true';
+    const hadPrevFreeTrial = subscription?.is_free_trial === 'false' && subscription?.trial_end_at > 0;
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+
     let daysLeftOnTrial = getRemainingDaysFromFutureTimestamp(subscription?.trial_end_at);
     if (daysLeftOnTrial > TrialPeriodDays.TRIAL_MAX_DAYS) {
         daysLeftOnTrial = TrialPeriodDays.TRIAL_MAX_DAYS;
     }
 
-    const show = isCloud && isFreeTrial;
+    const nonCloudFreeButIsFreeTrial = !isCloudFreeEnabled && isFreeTrial;
+
+    const openTrialBenefitsModal = async () => {
+        await dispatch(openModal({
+            modalId: ModalIdentifiers.TRIAL_BENEFITS_MODAL,
+            dialogType: TrialBenefitsModal,
+        }));
+    };
+
+    const show = isCloud && (nonCloudFreeButIsFreeTrial || isCloudFreeEnabled);
     if (!show) {
         return null;
     }
 
-    return (
-        <li
-            className={'MenuCloudTrial'}
-            role='menuitem'
-            id={id}
-        >
+    // this is the menu content displayed when the workspace is running a trial. It would display a different option depending if
+    // cloudFree is enabled or not.
+    const freeTrialContent = isCloudFreeEnabled ? (
+
+        // Here go the limits calculation and refactor of the menu option
+        <>
+            <FormattedMessage
+                id='menu.cloudFree.tryEnterprise'
+                defaultMessage='Are you making the most of your Enterprise trial? '
```

Comment:
Looks like we need to create the id for this message. Right now, it is accidentally displaying the `Interested in a limitless plan with high-security features?` message.

![needs-id](https://user-images.githubusercontent.com/13738432/167892254-a15d480c-7e45-4715-a9fd-eb3b527bc092.png)

Reply:
got it, nice catch!